File: Java/src/main/java/QuestionBank/Solution701_800/Solution796.java
package QuestionBank.Solution701_800;

/**
 * @author windows
 */
public class Solution796 {
    public boolean rotateString(String s, String goal) {
        if (s.length() != goal.length()){
            return false;
        }
        s += s;
        return s.indexOf(goal) != -1;
    }


    public static void main(String[] args) {
        Solution796 solution796 = new Solution796();
    }
}

File: C++/src/Solution701_800/Solution796.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution796.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

bool rotateString(string s, string goal) {
  if (s.size() != goal.size()) return false;
  goal += goal;
  return goal.find(s) != string::npos;
}